sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/CredentialsServiceImpl.java

```diff
@@ -45,7 +48,34 @@ public void reinitPassword(
       PasswordChangeRequest pcr,
       List<SendMode> sendMode) {
     User user = storeProvider.getReaderStore(realm, userStorage).getUser(userId);
-    String password = passwordService.generatePassword();
+    Map<String, String> usProperties =
+        userStorage != null
+            ? configService.getRealm(realm).getUserStorages().stream()
+                .filter(u -> u.getName().equalsIgnoreCase(userStorage))
+                .findFirst()
+                .get()
+                .getProperties()
+            : configService.getRealm(realm).getUserStorages().stream()
+                .findFirst()
+                .get()
+                .getProperties();
```

Comment:
What happens if I pass a null userStorage and the user exist in another userStorage than the one chosen here ?

(Imagine we have realm1 with 2 us (us1 and us2) I'm trying to reinit password for user1 in us2 but the usProperties here is taken from us1, can this happen ?

Reply:
If we pass a null userstorage and we do a getreaderstore, it will be considered that we are doing the action on the first userstorage (for the moment maybe it will change in the futur). That's why I do a findfirst when I try to get usproperties. I expected to get the same userstorage between the first operation and the second one. Maybe I'm wrong ^^